src/platform_posix.cc

```diff
@@ -368,4 +368,28 @@ optional<std::string> RunExecutable(const std::vector<std::string>& command,
   return result;
 }
 
+optional<std::string> GetGlobalConfigDirectory() {
+  char const* xdg_config_home = std::getenv("XDG_CONFIG_HOME");
+  char const* home = std::getenv("HOME");
+  std::string config;
+
+  // If it exists use XDG_CONFIG_HOME to comply with the XDG base
+  // directory spec. Otherwise, use HOME if available.
+  if (xdg_config_home) {
+    config = xdg_config_home;
+    EnsureEndsInSlash(config);
+    config += "cquery";
+    if (!FileExists(config)) {
+      MakeDirectoryRecursive(AbsolutePath(config, true));
+    }
+  } else if (home) {
+    config = home;
+  } else {
+    return {};
```

Comment:
nit: `return nullopt;`

But instead of having a separate return here (returns in the middle of a func make control-flow hard to read) you can refactor this function so that `config` is an optional. The final call to `EnsureEndsInSlash` will need to be `if (config) EnsureEndsInSlash(*config);`

Reply:
The refactor definitely makes it cleaner, have done as you recommended